src/main/java/org/folio/service/RequestExpirationService.java

```diff
@@ -50,29 +53,28 @@ public class RequestExpirationService {
   private final Function<Request, String> requestClassifier;
   private final PostgresClient pgClient;
   private final EventPublisherService eventPublisherService;
+  private final EntityChangedEventPublisher<String, Request> eventPublisher;
 
   public RequestExpirationService(Map<String, String> okapiHeaders, Vertx vertx,
-    String requestClassifierProperty, Function<Request, String> requestClassifier) {
+    Context vertxContext, String requestClassifierProperty,
+    Function<Request, String> requestClassifier) {
 
     this.requestClassifierProperty = requestClassifierProperty;
     this.requestClassifier = requestClassifier;
     pgClient = PostgresClient.getInstance(vertx, okapiHeaders.get(TENANT_HEADER));
     eventPublisherService = new EventPublisherService(vertx, okapiHeaders);
+    this.eventPublisher = requestEventPublisher(vertxContext, okapiHeaders);
   }
 
-  @SuppressWarnings("java:S1905")  // suppress false positive "Remove this unnecessary cast to Void"
   public Future<Void> doRequestExpiration() {
-    List<JsonObject> context = new ArrayList<>();
+    List<ExpiredRequestWrapper> context = new ArrayList<>();
 
     return pgClient.withTrans(conn -> getExpiredRequests(conn)
         .compose(expiredRequests -> closeRequests(conn, expiredRequests, context))
         .compose(associatedIds -> getOpenRequestsByIdFields(conn, associatedIds))
         .compose(openRequests -> reorderRequests(conn, openRequests))
-        .map(x -> {
-          context.forEach(p -> eventPublisherService
-              .publishLogRecord(new JsonObject().put(REQUESTS.value(), p), REQUEST_EXPIRED));
-          return (Void)null;
-        }))
+        .map(x -> publishPubSubLogEvents(context)))
```

Comment:
```suggestion
        .onSuccess(x -> publishPubSubLogEvents(context)))
```
This way you can avoid the need to return a value from this method.